### DISCUSSION TITLE
Direct Dowload Link = Error 500 Internal Server Error

### DISCUSSION THREAD

**Discussion by depiuri** [spencerwooo/onedrive-vercel-index#477] [Q&A] [answered]
Files I uploaded after Feb 18, their direct download links throw this error, but downloading from index works.
https://i.imgur.com/qk2Um12.png

**Comment by myl7** [comment]
Try to upgrade to the latest version if you are not using it.
And could you give a full URL that would reproduce the above error?

**Reply by depiuri** [reply]
I've been trying, reinstall a database, but I keep getting the error with only some files. https://imgur.com/a/YkzZORp

**Comment by spencerwooo** [comment, marked as answer]
Latest version changed raw direct download paths to `/api/raw/?path=<path>` instead of `/api/?path=<path>&raw=true`, try the new link? The old one is only kept for backwards compatibility.

**Reply by depiuri** [reply]
I already tried it, update the new version, I made another fork, reviewing the documentation, try with different types of URLs. But nothing.

**Reply by spencerwooo** [reply]
No I don't think you updated to the latest version, API on your site still contains this `@microsoft.graph.downloadUrl` - `https://public.dm.files.1drv.com`, which is not used in the latest release.

![CleanShot 2022-02-22 at 13 45 01](https://user-images.githubusercontent.com/32114380/155070529-f615a8e3-4de3-449a-b226-4325d554975b.png)

**Reply by depiuri** [reply]
I forked 5 minutes ago from the array. C: I already saw what generated the error, it was the name of my file that had a "+" in the name, I removed it and it works, even so I will see what I did wrong in the update. Thank you very much for your attention.

**Reply by myl7** [reply]
`+` problem should be fixed in #465 . Anyway, the direct download url will be fixed later to make it almostly always work. See discussion #483 .
